src/full/Agda/TypeChecking/Sort.hs

```diff
@@ -101,22 +101,21 @@ inferPiSort a s2 = do
   case piSort' (unEl <$> a) s1' s2' of
     Right s -> return s
     Left b -> do
-      let b' = unblockOnAllMetasIn (s1' ,unAbs s2')
-      addConstraint (unblockOnBoth b b') $ HasPTSRule a s2'
+      addConstraint b $ HasPTSRule a s2'
```

Comment:
Here you should also take into account the blocking tags coming from the two calls to `reduce` above (which should be `reduceB` instead)